src/chrdev.rs

```diff
@@ -141,12 +164,36 @@ unsafe extern "C" fn release_callback<T: FileOperations>(
     0
 }
 
+unsafe extern "C" fn llseek_callback<T: FileOperations>(
+    file: *mut bindings::file,
+    offset: bindings::loff_t,
+    whence: c_types::c_int,
+) -> bindings::loff_t {
+    let off = match whence as u32 {
+        bindings::SEEK_SET => {
+            if offset < 0 {
+                return Error::EINVAL.to_kernel_errno().into();
+            }
+            SeekFrom::Start(offset.try_into())
+        }
+        bindings::SEEK_CUR => SeekFrom::Current(offset),
+        bindings::SEEK_END => SeekFrom::End(offset),
+        _ => return Error::EINVAL.to_kernel_errno().into(),
+    };
```

Comment:
How do you feel about this? I think something sorta like this should work:
```rust
    let off = match match whence as u32 {
        bindings::SEEK_SET => offset.try_into().map(SeekFrom::Start).map_err(From::from)
        bindings::SEEK_CUR => Ok(SeekFrom::Current(offset.into())),
        bindings::SEEK_END => Ok(SeekFrom::End(offset.into()),
        _ => Error::EINVAL,
    } {
        Ok(x) => x,
        Err(e) => { return e.to_kernel_errno().into(); }
    }
```

(It occurs to me we'll need the `.into()`s for the 32-bit build, but I'm not sure I care yet, the whole thing is probably not 32-bit clean.)

Happy to hold off until #60, my proposal there would let you write an extern fn that you can use `?` on.

Reply:
Yeah, I think I'd prefer to wait until #60 or something like it is landed, I think that's less readable than what we have currently.